storage/mysql/mysql.go

```diff
@@ -99,17 +99,44 @@ func (s *Storage) writeCheckpoint(ctx context.Context, rawCheckpoint []byte) err
 }
 
 // ReadTile returns a full tile or a partial tile at the given level and index.
-func (s *Storage) ReadTile(ctx context.Context, level, index uint64) ([]byte, error) {
+//
+// TODO: Handle the following scenarios:
+// 1. Full tile request with full tile output: Return full tile.
+// 2. Full tile request with partial tile output: Return error.
+// 3. Partial tile request with full/larger partial tile output: Return trimmed partial tile with correct tile width.
```

Comment:
Worth just thinking about returning the full tile/bundle in the cases where the request is for a smaller partial size...

Clients of the log should only be requesting tiles when they have a checkpoint "in hand" which implies their existence. They should never rely on anything beyond the bounds of the tree which the checkpoint commits to since anything there is by definition not committed to. So, there shouldn't really be any issue if returning a larger/full tile when a partial one was requested.

Reply:
Quite right. Let's have a team discussion this week.